captum/metrics/_core/sensitivity.py

```diff
@@ -0,0 +1,256 @@
+
+            Default: 0.02
+        n_samples (int, optional): The number of times input tensors are perturbed.
+                Each input example in the inputs tensor is expanded `n_samples`
+                times before calling `perturb_func` function.
+
+                Default: 10
+        norm_ord (str, optional): The type of norm that is used to compute the
+                norm of the sensitivity matrix which is defined as the difference
+                between the explanation function at its input and perturbed input.
+
+                Default: "fro"
+        max_examples_per_batch (int, optional): The number of maximum input
+                examples that are processed together. In case the number of
+                examples (`input batch size * n_samples`) exceeds
+                `max_examples_per_batch`, they will be sliced
+                into batches of `max_examples_per_batch` examples and processed
+                in a sequential order. If `max_examples_per_batch` is None, all
+                examples are processed together.
+
+                Default: None
+         **kwargs (Any, optional): Contains a list of arguments that are passed
+                to `explanation_func` explanation function which in some cases
+                could be the `attribute` function of an attribution algorithm.
+                Any additional arguments that need be passed to the explanation
+                function should be included here.
+                For instance, such arguments include:
+                `additional_forward_args`, `baselines` and `target`.
+
+    Returns:
+
+        sensitivities (tensor): A tensor of scalar sensitivity scores per
+               input example. The first dimension is equal to the
+               number of examples in the input batch and the second
+               dimension is one.
+
+    Examples::
+        >>> # ImageClassifier takes a single input tensor of images Nx3x32x32,
+        >>> # and returns an Nx10 tensor of class probabilities.
+        >>> net = ImageClassifier()
+        >>> saliency = Saliency(net)
+        >>> input = torch.randn(2, 3, 32, 32, requires_grad=True)
+        >>> # Computes sensitivity score for saliency maps
+        >>> sens = sensitivity_max(net, saliency.attribute)
+
+    """
+
+    def _generate_perturbations(current_n_samples):
+        r"""
+        The perturbations are generated for each example `current_n_samples` times.
+
+        For perfomance reasons we are not calling `perturb_func` on each example but
+        on a batch that contains `current_n_samples` repeated instances per example.
+        """
+        inputs_expanded = tuple(
+            torch.repeat_interleave(input, current_n_samples, dim=0) for input in inputs
+        )
+        if len(inputs_expanded) == 1:
+            inputs_expanded = inputs_expanded[0]
+
+        return (
+            perturb_func(inputs_expanded, perturb_radius)
+            if perturb_func == default_perturb_func
```

Comment:
I was playing with different options  and in this case since we pass tuple of tensors in case of multiple inputs, we can check the number of arguments and always pass the perturbation radius. Nevertheless it is not an ideal solution as well, because the user might chose to pass something else instead of `perturb_radius` as second argument.